cms/static/js/views/move_xblock_list.js

```diff
@@ -0,0 +1,213 @@
+            } else if (direction === 'forward') {
+                // clicked child is the new parent
+                this.parentInfo.parent = this.childrenInfo.children[newParentIndex];
+            } else if (direction === 'backward') {
+                // new parent will be one of visitedAncestors
+                this.parentInfo.parent = this.visitedAncestors[newParentIndex];
+                // remove visited ancestors
+                this.visitedAncestors.splice(newParentIndex);
+            }
+
+            this.visitedAncestors.push(this.parentInfo.parent);
+
+            if (this.parentInfo.parent.get('child_info')) {
+                this.childrenInfo.children = this.parentInfo.parent.get('child_info').children;
+            } else {
+                this.childrenInfo.children = [];
+            }
+
+            this.setDisplayedXBlocksCategories();
+            this.render();
+        },
+
+        /**
+         * Set parent and childs XBlock categories.
+         */
+        setDisplayedXBlocksCategories: function() {
+            this.parentInfo.category = XBlockUtils.getXBlockType(
+                this.parentInfo.parent.get('category'),
+                this.visitedAncestors[this.visitedAncestors.length - 2]
+            );
+            this.childrenInfo.category = this.categoryRelationMap[this.parentInfo.category];
+        },
+
+        /**
+         * Get index of source XBlock.
+         *
+         * @returns {any} Integer or undefined
+         */
+        getCurrentLocationIndex: function() {
+            var category, ancestorXBlock, currentLocationIndex;
+
+            if (this.childrenInfo.category === 'component' || this.childrenInfo.children.length === 0) {
+                return currentLocationIndex;
+            }
+
+            category = this.childrenInfo.children[0].get('category');
+            ancestorXBlock = _.find(
+                this.ancestorInfo.ancestors, function(ancestor) { return ancestor.category === category; }
+            );
+
+            if (ancestorXBlock) {
+                _.each(this.childrenInfo.children, function(xblock, index) {
+                    if (ancestorXBlock.display_name === xblock.get('display_name') &&
+                        ancestorXBlock.id === xblock.get('id')) {
+                        currentLocationIndex = index;
+                    }
+                });
+            }
+
+            return currentLocationIndex;
+        },
+
+        /**
+         * Get category text for currently displayed children.
+         *
+         * @returns {String}
+         */
+        getCategoryText: function() {
+            return this.categoriesText[this.childrenInfo.category];
+        },
+
+        /**
+         * Get screen reader text for forward button.
+         *
+         * @returns {String}
+         */
+        getForwardButtonSRText: function() {
+            return StringUtils.interpolate(
+                gettext('Press button to see {XBlockCategory} children'),
```

Comment:
Nit: this string seems like it needs work. Some current examples:

    Press button to see section children
    Press button to see subsection children
    Press button to see unit children

Also, I don't know that screen reader text needs to say "Press button" on a button. I would think this would be implied by the context.

How about: "View this section's children"? @catong, what do you think?